src/test/java/net/sf/jabref/logic/bibtex/BibEntryWriterTest.java

```diff
@@ -62,6 +62,71 @@ public void testSerialization() throws IOException {
     }
 
     @Test
+    public void readOtherTypeTest() throws IOException {
+        String bibtexEntry = "@Other{test," + OS.NEWLINE +
+                "  Comment                  = {testentry}" + OS.NEWLINE +
+                "}";
+
+        // read in bibtex string
+        ParserResult result = BibtexParser.parse(new StringReader(bibtexEntry), importFormatPreferences);
+        Collection<BibEntry> entries = result.getDatabase().getEntries();
+        BibEntry entry = entries.iterator().next();
+
+        String resourceName = "/testbib/othertype.bib";
+        BibEntryAssert.assertEquals(BibEntryWriter.class, resourceName, entry);
+    }
+
+    @Test
+    public void roundTripOtherTypeTest() throws Exception {
+        String bibtexEntry = "@Other{test," + OS.NEWLINE +
+                "  Comment                  = {testentry}" + OS.NEWLINE +
+                "}";
+
+        // read in bibtex string
+        Collection<BibEntry> entries = new BibtexParser(importFormatPreferences).parseEntries(bibtexEntry);
+        BibEntry entry = entries.iterator().next();
+
+        //write out bibtex string
+        StringWriter stringWriter = new StringWriter();
+        writer.write(entry, stringWriter, BibDatabaseMode.BIBTEX);
+        String actual = stringWriter.toString();
+
+        assertEquals(bibtexEntry, actual);
+    }
+
+    @Test
+    public void readReallyUnknownTypeTest() throws Exception {
+        String bibtexEntry = "@ReallyUnknownType{test," + OS.NEWLINE +
```

Comment:
If you have a test for reading something, then you should place it in the ParserTest ;-)